src/pymor/algorithms/qr.py

```diff
@@ -0,0 +1,67 @@
+# This file is part of the pyMOR project (https://www.pymor.org).
+# Copyright pyMOR developers and contributors. All rights reserved.
+# License: BSD 2-Clause License (https://opensource.org/licenses/BSD-2-Clause)
+
+from pymor.algorithms.gram_schmidt import gram_schmidt
+
+
+def qr(A, product=None, offset=0, check=True, check_tol=1e-3, copy=True):
```

Comment:
- Add `method` argument to choose between implementations. Make it a default.
- Add `**kwargs` to be able to pass through implementation specific arguments.
- Make `check` and `check_tol` defaults.